### PR TITLE
Nodejs/week3

### PR DESCRIPTION
**Pull request by dinythomas89**

### CODE REVIEW

File: src/backend/api/meals.js [hunk 1]
```diff
@@ -4,9 +4,132 @@ const knex = require("../database");
...
+    //to get meals that partially match a title.
+    if ('title' in request.query) {
+      const title = request.query.title;
+      const mealWithGivenTitle = await knex("meal").where('title', 'like', "%" + title + "%");
+      response.send(mealWithGivenTitle);
+    }
+
+    //to get meals that has been created after the date
+    if ('createdAfter' in request.query) {
+      const createdAfter = request.query.createdAfter;
```
Comment (flom84) [inline src/backend/api/meals.js:48]:
```suggestion
      const createdAfter = Date.parse(request.query.createdAfter);
```
when you have query params make sure you parse the data and check if it's correct

File: src/backend/api/meals.js [hunk 2]
```diff
@@ -4,9 +4,132 @@ const knex = require("../database");
...
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    //to get meals that has a price smaller than maxPrice
+    if ('maxPrice' in request.query) {
+      const maxPrice = parseInt(request.query.maxPrice);
+      if (isNaN(maxPrice)) {
+        response.status(400).send({ error: "maxPrice should be an integer" });
+        return;
+      }
+      const mealsLessThanMaxPrice = await knex("meal").where('price', '<', maxPrice);
```
Comment (flom84) [inline src/backend/api/meals.js:14]:
```suggestion
      const mealsLessThanMaxPrice = await knex("meals").where('price', '<', maxPrice);
```
i think the name of the database is `meals` and not `meal`

File: src/backend/api/meals.js [hunk 3]
```diff
@@ -4,9 +4,132 @@ const knex = require("../database");
...
+  }
+  catch (error) {
+    throw error;
+  }
+});
+
+//Adds a new meal
+router.post("/", async (request, response) => {
+  try {
+    const meals = await knex("meal").insert(request.body);
```
Comment (flom84) [inline src/backend/api/meals.js:80]:
```suggestion
    const meal = await knex("meals").insert(request.body);
```

File: src/backend/api/meals.js [hunk 4]
```diff
@@ -4,9 +4,132 @@ const knex = require("../database");
...
+    response.json(meals);
```
Comment (flom84) [inline src/backend/api/meals.js:81]:
```suggestion
    response.json(meal);
```

**Review by flom84 — CHANGES_REQUESTED** [review]
I think the name of the database should be `meals` and not `meal`.

File: src/backend/api/meals.js (hunk 4 above)
Comment (dinythomas89) [reply src/backend/api/meals.js:81]: Thanks Florin. will implement the changes as you mentioned..

File: src/backend/api/meals.js (hunk 2 above)
Comment (dinythomas89) [reply src/backend/api/meals.js:14]: hi florin, the table name is meal

**Pull request merged**